dxp/demystifying_aui_script_and_liferay-frontend_component_tags.md

```diff
@@ -0,0 +1,114 @@
+# Demystifying `aui:script` and `liferay-frontend:component` tags
+
+On JSP context you may notice a huge number of `aui:script` and `liferay-frontend:component` tags.
+
+## `aui:script` tag
+
+### What it does:
+
+This taglib allow users to use and require another JS module/resources inside a JSP.
+
+This taglib have three possible use cases:
+
+1. Pure JS on the JSP
+2. Import a AUI/YUI module
+3. Import a ES6 module
+
+#### Pure JS on the JSP
```

Comment:
I'd add an example to this use case

Reply:
> Again here I'd drop "Pure" and say "Inline" (or "embedded")?

I got you thoughts here. I would prefer use `inline` since embedded can be interpret wrongly as a way to embed an external JS within the JSP body.

Reply:
> I'd add an example to this use case

I'll add an example now

Reply:
Does it looks good? I would go for a simpler and maybe practical use case that involves Java too

```jsp
<aui:script>
	Liferay.fire('myGlobalEvent', {
		'payload': <%= myDisplayContext.getPayload() %>
	});
</aui:script>
```

Reply:
I would even drop the `var` to make it brief (just inline it); eg. `Liferay.fire('some-event'`

The thing about changing that JSP expression from `<%= myDisplayContext.getPayload() %>` to `'myPayload'` is that you're blurring the lines between what the `<aui:script>` tag itself does, and how JSP itself works in general. This level of detail is extraneous. I know you want to be helpful, but by giving more information you're making your message less clear. This really could be as simple as:

> This example:
> 
> ```
> <aui:script>
>     // some JS code...
> </aui:script>
> ```
> 
> will be turned into:
> 
> ```js
> <script>
>     // some JS code...
> </script>
> ```

(Note: you should check what the output actually is. I don't think you get the IIFE without `sandbox`.)